src/main.rs

```diff
@@ -191,17 +191,18 @@ fn display_wallet_balance(wallet_file_name: &PathBuf, long_form: Option<bool>) {
                     compressed: true,
                     key: bitcoin::secp256k1::PublicKey::from_secret_key(
                         &secp,
-                        &swapcoin.contract_privkey,
+                        swapcoin.get_contract_privkey().unwrap(),
                     ),
                 };
                 let type_string = if contract_pubkey
-                    == read_hashlock_pubkey_from_contract(&swapcoin.contract_redeemscript).unwrap()
+                    == read_hashlock_pubkey_from_contract(&swapcoin.get_contract_redeemscript())
+                        .unwrap()
                 {
                     "hashlock"
                 } else {
                     assert_eq!(
                         contract_pubkey,
-                        read_timelock_pubkey_from_contract(&swapcoin.contract_redeemscript)
+                        read_timelock_pubkey_from_contract(&swapcoin.get_contract_redeemscript())
                             .unwrap()
                     );
                     "timelock"
```

Comment:
As "timelock" and "hashlock" exactly match the OutgoingSwapCoin and IncomingSwapCoin types, we no longer need to do this whole part where we calculate the public key and compare it with the contract_redeemscript. Consequently we also don't need the function `get_contract_privkey`.

Reply:
I replaced these tests with a function `SwapCoin::contract_type` that directly returns the type string:

- IncomingSwapCoin => "hashlock"
- OutgoingSwapCoin => "timelock"
- WatchOnlySwapCoin => "watchonly"